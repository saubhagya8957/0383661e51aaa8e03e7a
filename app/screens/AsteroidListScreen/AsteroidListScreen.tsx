import React, { Component } from "react";
import { View, SafeAreaView, Text, TextInput, TouchableOpacity, Alert, FlatList } from 'react-native';
import styles from './AsteroidListScreenStyle';


interface Props {
    navigation: any;
    id: string;
    route: any;
}

interface S {
    asteroidData: any;
    asteroidDataList: any;
}

interface SS {
    id: any
}

class AsteroidListScreen extends Component<Props, S, SS> {
    constructor(props: Props) {
        super(props);
        this.state = {
            asteroidData: this.props.route.params.asteroidDataList,
            asteroidDataList: [],
        }
    }

    componentDidMount = () => {
        console.log('@@@ Asteroid Params Response ===========', this.props.route.params.asteroidData);
    }

    componentWillUnmount = () => {

    }

    renderAsteroidCell = (item: any, index: any) => {
        return (
            <TouchableOpacity onPress={() => this.props.navigation.navigate('AsteroidDetailScreen', { asteroidDetail: item })} style={styles.asteroidDataCell}>
                <Text style={styles.key}>Name: {item.name}</Text>
                <Text style={styles.key}>NASA JPL URL: {item.nasa_jpl_url}</Text>
                <Text style={styles.key}>IS POTENTIALLY HAZARDOUS ASTEROID: {String(item.is_potentially_hazardous_asteroid)}</Text>
            </TouchableOpacity>
        )
    }

    render() {
        return (
            <SafeAreaView style={styles.container}>
                <FlatList
                    data={this.state.asteroidData}
                    renderItem={({ item, index }) => this.renderAsteroidCell(item, index)}
                    keyExtractor={(item) => item.id}
                    extraData={this.state}
                />
            </SafeAreaView>
        );
    }
}

export default AsteroidListScreen;