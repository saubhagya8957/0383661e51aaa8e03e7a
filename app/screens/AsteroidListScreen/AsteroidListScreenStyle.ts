import { StyleSheet } from "react-native";


export default StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#ab8e00'
    },

    asteroidDataCell: {
        width: 350,
        alignItems: 'center',
        backgroundColor: '#628ead',
        paddingVertical: 10,
        borderColor: '#000000',
        paddingRight: 12,
        alignSelf: 'center',
        marginBottom: 10
    },

    key: {
        fontSize: 16,
        color: '#ffffff',
        fontWeight: 'bold',
        marginLeft: 20,
        marginVertical: 10,
        width: 300
    }
})  