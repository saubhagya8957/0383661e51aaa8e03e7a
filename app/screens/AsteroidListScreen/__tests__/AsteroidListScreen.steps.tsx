import React from "react";
import { defineFeature, loadFeature } from 'jest-cucumber';
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AsteroidListScreen from "../AsteroidListScreen";

configure({ adapter : new Adapter() });

const screenProps = {
    navigation: {
        navigate: jest.fn(),
    },
    route: {
        params: {
            asteroidDataList: []
        }
    },
    id: 'Asteroid Data List'
}

const feature = loadFeature('./app/screens/AsteroidListScreen/__tests__/features/AsteroidListScreen.scenarios.feature');

defineFeature(feature, (test) => {
    test('User navigates Asteroid List', ({ given, when, then }) => {
        let AsteroidListScreenWrapper: ShallowWrapper;
        let AsteroidListScreenInstance: AsteroidListScreen;

        given('I am user loading Asteroid List Screen', () => {
            AsteroidListScreenWrapper = shallow(<AsteroidListScreen {...screenProps} />);
        });

        when('I just navigate to the Asteroid List Screen', () => {
            AsteroidListScreenInstance = AsteroidListScreenWrapper.instance() as AsteroidListScreen;
        });

        then('Asteroid List Screen will mount with out any errors', () => {
            expect(AsteroidListScreenWrapper).toBeTruthy();
            expect(AsteroidListScreenWrapper).toMatchSnapshot();
        });

        then('I can unmount the screen with out any errors', () => {
            AsteroidListScreenInstance.componentWillUnmount();
            expect(AsteroidListScreenWrapper).toBeTruthy();
            expect(AsteroidListScreenWrapper).toMatchSnapshot();
        });
    })
})