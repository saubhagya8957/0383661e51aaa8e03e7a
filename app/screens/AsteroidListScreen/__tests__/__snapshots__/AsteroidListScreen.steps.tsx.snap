// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asteroid List Screen User navigates Asteroid List 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    extraData={
      Object {
        "asteroidData": Array [],
        "asteroidDataList": Array [],
      }
    }
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</ForwardRef(SafeAreaView)>
`;

exports[`Asteroid List Screen User navigates Asteroid List 2`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    extraData={
      Object {
        "asteroidData": Array [],
        "asteroidDataList": Array [],
      }
    }
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</ForwardRef(SafeAreaView)>
`;
